File: info/htmlclasses/HtmlElement.php
<?php 
    class Attribute {
        private $name;
        private $value;
        public function __construct($name,$value) {
            $this->name = $name;
            $this->value = $value;
        }
        public function getName() {
            return $this->name;
        }
        public function getValue() {
            return $this->value;
        }
    }
    class TextElement {
        private $text;

        public function __construct($text){
            $this->name = $text;
        }
        public function render(){
            return $this->text;
        }
    }
    class HtmlElement extends TextElement{
        protected $name;

        public function __construct($name){
            $this->name = $name;
        }
        public function render(){
            return $this->name;
        }
    }
    class WithAttribute extends HtmlElement {
        protected $attributes = array();

        public function addAttribute($attribute) {
            $this->attributes = $attribute;
        }
        // incomplete
        public function render() {
            foreach($this->attributes as $x => $x_value) {
                echo "Key=" . $x . ", Value=" . $x_value;
                echo "<br>";
            }
        }
    }
    $ad = new WithAttribute($name);
?>

File: info/htmlclasses/HtmlElement.py
class Attribute:
        __name = ""
        __value = ""
        def __init__(self,name,value):
            self.__name = name
            self.__value = value
        def getName(self):
            return self.__name;
        def getValue(self): 
            return self.__value;

class TextElement:
    __text = ""
    def __init__(self,text):
        self.__text = text
    def render(self):
        return self.__text

class HtmlElement(TextElement):
    _name = ""
    def __init__(self, name):
        self._name = name
    def render(self):
        return "<"+self._name+">"

class WithAttribute(HtmlElement):
    _attributes = {}
    def addAttribute(self, attribute):
        self._attributes[attribute.getName()] = attribute.getValue()
    def render(self):
        out = ""
        for key,val in self._attributes:
            out.append(" "+key+"="+val)
        return "<"+self._name+out+">"

class WithContent(WithAttribute):
    _content = ""
    def addContent():
        